### ISSUE TITLE
halfspace_intersection_3 fails on following sample data (after sloriot fix)

### ISSUE THREAD

**Issue by bombax** [CGAL/cgal#112]
After applying fix https://github.com/CGAL/cgal/pull/82 to my program with massive sample data, there were unfortunately large amounts of fails of the kind "origin not in polyhedron" when using boost::none.

I can give more samples on request.

(@sloriot)

The following illustrates the first 6 of the failures:

```cpp
#include <iostream>
#include <list>
#include <cstdint>
#include <cstdlib>
#include <exception>

#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Convex_hull_3/dual/halfspace_intersection_3.h>

using namespace std;

typedef CGAL::Exact_predicates_inexact_constructions_kernel Kernel;
typedef Kernel::Plane_3                                     Plane;
typedef Kernel::Point_3                                     Point;
typedef CGAL::Polyhedron_3<Kernel>                          Polyhedron;

int planeCount[6] = {7, 7, 8, 8, 8, 7};
uint64_t planeBits[6][8][4] = {
{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0x4092000000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0xc096800000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0x40a3200000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0xc0a3400000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0x4070000000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc080000000000000ull},
    {0x3fe6a09e60000000ull, 0xbfe6a09e60000000ull, 0x0ull, 0x4086a09e60000000ull}
},{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0xc08a800000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0x4088800000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0x4092800000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0xc094200000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0xc07a000000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc090800000000000ull},
    {0x3feb40c900000000ull, 0x3fe0c56800000000ull, 0x0ull, 0x4047959a40000000ull}
},{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0xc068000000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0xc068000000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0xc094000000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0x4090000000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0xc058000000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc05a000000000000ull},
    {0xbfefc0bd80000000ull, 0x3fbfc0bd80000000ull, 0x0ull, 0xc03fc0bd80000000ull},
    {0x3fefc0bd80000000ull, 0x3fbfc0bd80000000ull, 0x0ull, 0xc03fc0bd80000000ull}
},{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0x4071c00000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0xc089a00000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0x4075c00000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0xc076400000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0x4071800000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc072800000000000ull},
    {0xbfe6a09e60000000ull, 0x3fe6a09e60000000ull, 0x0ull, 0xc046a09e60000000ull},
    {0x3fe6a09e60000000ull, 0x3fe6a09e60000000ull, 0x0ull, 0xc089cf34a0000000ull}
},{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0xc0b5200000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0x40b3600000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0x40aa380000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0xc0aa400000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0x4071000000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc072000000000000ull},
    {0x3fe6a09e60000000ull, 0xbfe6a09e60000000ull, 0x0ull, 0x40b6fb20e0000000ull},
    {0xbfe6a09e60000000ull, 0xbfe6a09e60000000ull, 0x0ull, 0xc096a09e60000000ull}
},{
    {0xbff0000000000000ull, 0x0ull, 0x0ull, 0x4092000000000000ull},
    {0x3ff0000000000000ull, 0x0ull, 0x0ull, 0xc096800000000000ull},
    {0x0ull, 0xbff0000000000000ull, 0x0ull, 0x40a3200000000000ull},
    {0x0ull, 0x3ff0000000000000ull, 0x0ull, 0xc0a3400000000000ull},
    {0x0ull, 0x0ull, 0xbff0000000000000ull, 0x4070000000000000ull},
    {0x0ull, 0x0ull, 0x3ff0000000000000ull, 0xc080000000000000ull},
    {0x3fe6a09e40000000ull, 0xbfe6a09e40000000ull, 0x0ull, 0x4086a09e40000000ull}
}

};

int main(int argc, char *argv[]) {
    int choice = 3; // choose between [0, 6)

    list<Plane> planes;
    for (int i = 0; i < planeCount[choice]; i++) {
        double a, b, c, d;
        memcpy(&a, &planeBits[choice][i][0], 8);
        memcpy(&b, &planeBits[choice][i][1], 8);
        memcpy(&c, &planeBits[choice][i][2], 8);
        memcpy(&d, &planeBits[choice][i][3], 8);

        planes.push_back(Plane(a, b, c, d));
    }

    try {
        Polyhedron poly;
        CGAL::halfspace_intersection_3(planes.begin(),
                                       planes.end(),
                                       poly,
                                       static_cast<boost::optional<Point> >(boost::none));
    } catch (exception &e) {
        cerr << e.what() << endl;
        cerr << "for planes" << endl;
        for (auto it = planes.begin(); it != planes.end(); it++)
            cerr << *it << endl;
        cin.get();
    }

    return 0;
}
```

**Comment by sloriot** [comment]
I can reproduce the issue with master but #116 seems to fix the issue.

**Comment by lrineau** [comment]
Good. It means that this bug will be soon be fixed.

**Comment by bombax** [comment]
Excellent. I will test all my sample data when a fix is available.

**Comment by lrineau** [comment]
@sloriot, @bombax: Is this bug fixed or not?

**Comment by lrineau** [comment]
Seems fixed by #116

**Comment by bombax** [comment]
After applying fix https://github.com/CGAL/cgal/pull/116 there are unfortunately still a few (though not as many) failures with halfspace_intersection_3, see issue here: https://github.com/CGAL/cgal/issues/200
